src/array_partition.jl

```diff
@@ -107,18 +107,25 @@ function Base.copyto!(dest::Array,A::ArrayPartition)
     @assert length(dest) == length(A)
     cur = 1
     @inbounds for i in 1:length(A.x)
-        dest[cur:(cur+length(A.x[i])-1)] .= A.x[i]
-        cur += length(A.x[i])
+      dest[cur:(cur+length(A.x[i])-1)] .= vec(A.x[i])
+      cur += length(A.x[i])
     end
     dest
 end
 
 function Base.copyto!(A::ArrayPartition,src::ArrayPartition)
     @assert length(src) == length(A)
-    cur = 1
-    @inbounds for i in 1:length(A.x)
-        A.x[i] .= @view(src[cur:(cur+length(A.x[i])-1)])
-        cur += length(A.x[i])
+    if size.(A.x) == size.(src.x)
```

Comment:
does this need `if all(...)`?

Reply:
It seems to be working without `all`... I guess when using `all` you would need `.==`?

Reply:
It seems we have to use `size.(A.x) == size.(src.x)`. If partition don't agree this can happen:
```julia
julia> x = ArrayPartition(randn(3,2),randn(2)); y = ArrayPartition(randn(3),randn(3),randn(2));

julia> all(size.(x.x) .== size.(y.x))
ERROR: DimensionMismatch("arrays could not be broadcast to a common size")
```